financial-connections/src/main/java/com/stripe/android/financialconnections/ui/FinancialConnectionsSheetNativeActivity.kt

```diff
@@ -270,34 +272,39 @@ internal class FinancialConnectionsSheetNativeActivity : AppCompatActivity(), Ma
     }
 
     @Composable
-    private fun NavigationEffect(navController: NavHostController) {
-        LaunchedEffect(navigationManager.commands) {
-            navigationManager.commands.collect { command ->
-                val from = navController.currentDestination?.route
-                if (command.destination.isNotEmpty() && command.destination != from) {
-                    logger.debug("Navigating from $from to ${command.destination}")
-                    navController.navigate(command.destination) {
-                        launchSingleTop = true
-                        popUpIfNotBackwardsNavigable(navController)
-                    }
+    private fun NavigationEffect(
+        navController: NavHostController
+    ) {
+        val navigationState by navigationManager.navigationState.collectAsState()
+
+        LaunchedEffect(navigationState) {
+            logger.debug("updateNavigationState to $navigationState")
+            val from = navController.currentDestination?.route
+            when (val viewState = navigationState) {
+                is NavigationState.NavigateToRoute -> {
+                    navigateToRoute(viewState, from, navController)
+                    navigationManager.onNavigated(navigationState)
                 }
+
+                is NavigationState.Idle -> {}
             }
         }
     }
 
-    /**
-     * Removes screens that are not backwards-navigable from the backstack.
-     */
-    private fun NavOptionsBuilder.popUpIfNotBackwardsNavigable(navController: NavHostController) {
-        val destination: String = navController.currentBackStackEntry?.destination?.route ?: return
-        val destinationsToSkipOnBack = listOf(
-            NavigationDirections.partnerAuth.destination,
-            NavigationDirections.reset.destination
-        )
-        if (destinationsToSkipOnBack.contains(navController.currentDestination?.route)
-        ) {
-            popUpTo(destination) {
-                inclusive = true
```

Comment:
The main goal of this PR was to get rid of this logic. Knowing the current screen and the navigation target is not enough to determine wether we should pop the current screen from the stack.

Even if it was, we'd me making pane-specific decisions on the parent activity, which does not scale.